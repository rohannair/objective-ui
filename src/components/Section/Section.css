.section {
  padding: 75px 40px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  color: #2E404E;
}

.sectionSplash {
  composes: section;
  padding: 150px 40px;
  color: #fff;
}

.heading {
  font-size: 2.25rem;
}

.splashHeading {
  font-size: 3rem;
  border-bottom: 1px solid #fff;
  padding: 0 100px 15px;
  margin-bottom: 15px;
}

.sub {
  font-size: 1.125rem;
  margin-bottom: 20px;
}

.splashSub {
  font-size: 2.25rem;
  padding-bottom: 20px;
}
