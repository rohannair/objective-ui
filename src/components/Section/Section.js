import React, { PropTypes } from 'react'
import styles from './Section.css'

const Section = props => {
  const textClasses = props.splash
  ? { heading: styles.splashHeading, sub: styles.splashSub }
  : { heading: styles.heading, sub: styles.sub }

  const sectionClasses = props.splash
  ? styles.sectionSplash
  : styles.section

  return (
    <div className={ sectionClasses } style= {{ backgroundColor: props.bgColor }}>
      <h2 className={ textClasses.heading }>{ props.title }</h2>
      <p className={ textClasses.sub }>{ props.subtitle }</p>
      <div>
        { props.children }
      </div>
    </div>
  )
}

Section.defaultProps = {
  bgColor: 'transparent',
  title: '',
  splash: false,
}

Section.propTypes = {
  bgColor: PropTypes.string,
  children: PropTypes.element,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  splash: PropTypes.bool,
}

export default Section
