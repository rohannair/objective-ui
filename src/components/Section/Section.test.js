import React from 'react'
import Section from './index.js'

describe('Test suite for Section component', () => {
  it('Section should exist', () => {
    let wrapper = shallow(<Section/>)
    expect(wrapper).toMatchSnapshot()
  })
})

