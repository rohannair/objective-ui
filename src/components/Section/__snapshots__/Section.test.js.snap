exports[`Test suite for Section component Section should exist 1`] = `
<div
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }>
  <h2 />
  <p />
  <div />
</div>
`;
